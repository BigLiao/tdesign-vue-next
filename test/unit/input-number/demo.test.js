import { mount } from '@vue/test-utils';
import center from '@/examples/input-number/demos/center.vue';
import defaultDemo from '@/examples/input-number/demos/default.vue';
import disabled from '@/examples/input-number/demos/disabled.vue';
import format from '@/examples/input-number/demos/format.vue';
import left from '@/examples/input-number/demos/left.vue';
import size from '@/examples/input-number/demos/size.vue';
import step from '@/examples/input-number/demos/step.vue';
import undefinedDemo from '@/examples/input-number/demos/empty.vue';

describe('InputNumber', () => {
  it('center demo works fine', () => {
    const wrapper = mount(center);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('defaultDemo demo works fine', () => {
    const wrapper = mount(defaultDemo);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('disabled demo works fine', () => {
    const wrapper = mount(disabled);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('format demo works fine', () => {
    const wrapper = mount(format);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('left demo works fine', () => {
    const wrapper = mount(left);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('size demo works fine', () => {
    const wrapper = mount(size);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('step demo works fine', () => {
    const wrapper = mount(step);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('undefined demo works fine', () => {
    const wrapper = mount(undefinedDemo);
    expect(wrapper.element).toMatchSnapshot();
  });
});
