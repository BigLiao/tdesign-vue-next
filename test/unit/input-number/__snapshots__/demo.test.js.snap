// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputNumber center demo works fine 1`] = `
<div>
  <div
    class="t-input-number t-size-m"
    modelvalue="3"
  >
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__decrease"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-remove t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128 480L896 480 896 544 128 544Z"
          fill-opacity=".9"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-input-number-text-align"
        placeholder=""
      />
    </div>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__increase"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-add t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M544 160L544 480 864 480 864 544 544 544 544 864 480 864 480 544 160 544 160 480 480 480 480 160 544 160Z"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
  </div>
</div>
`;

exports[`InputNumber defaultDemo demo works fine 1`] = `
<div>
  <div
    class="t-input-number t-size-m"
  >
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__decrease"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-remove t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128 480L896 480 896 544 128 544Z"
          fill-opacity=".9"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-input-number-text-align"
        placeholder=""
      />
    </div>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__increase"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-add t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M544 160L544 480 864 480 864 544 544 544 544 864 480 864 480 544 160 544 160 480 480 480 480 160 544 160Z"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
  </div>
</div>
`;

exports[`InputNumber disabled demo works fine 1`] = `
<div>
  <div
    class="t-input-number t-size-m t-is-disabled"
    modelvalue="3"
  >
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__decrease t-is-disabled"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-remove t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128 480L896 480 896 544 128 544Z"
          fill-opacity=".9"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-is-disabled t-input-number-text-align"
        disabled=""
        placeholder=""
      />
    </div>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__increase t-is-disabled"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-add t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M544 160L544 480 864 480 864 544 544 544 544 864 480 864 480 544 160 544 160 480 480 480 480 160 544 160Z"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
  </div>
</div>
`;

exports[`InputNumber format demo works fine 1`] = `
<div>
  <div
    class="t-input-number t-size-m"
    modelvalue="3"
  >
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__decrease"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-remove t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128 480L896 480 896 544 128 544Z"
          fill-opacity=".9"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-input-number-text-align"
        placeholder=""
      />
    </div>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__increase"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-add t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M544 160L544 480 864 480 864 544 544 544 544 864 480 864 480 544 160 544 160 480 480 480 480 160 544 160Z"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
  </div>
</div>
`;

exports[`InputNumber left demo works fine 1`] = `
<div>
  <div
    class="t-input-number t-size-m t-is-controls-right"
    modelvalue="3"
  >
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__decrease"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-chevron-down t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M226.57792 413.41696L285.40928 354.58624 511.9936 581.17056 738.5792 354.58624 797.408 413.41696 511.9936 698.8352 226.57792 413.41696Z"
          fill-opacity=".9"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner"
        placeholder=""
      />
    </div>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__increase"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-chevron-up t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M509.24008704 256.163392L514.71568448 256.13496384C521.97472 256.74779712 529.06494656 259.82337344 534.628192 265.36222272L534.628192 265.36222272 910.3463808 640.4937536 865.091552 685.7485888 511.99872896 333.24849216 160.00556288 685.7485888 114.75072896 640.4937536 466.75072896 288.06449216 466.72826816 288.00732608 489.35568512 265.37990912C493.52487168 261.22251328 498.55406784 258.45003712 503.87029504 257.06270464L509.24008704 256.163392 509.24008704 256.163392Z"
          fill-opacity=".9"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
  </div>
</div>
`;

exports[`InputNumber size demo works fine 1`] = `
<div>
  <div
    class="t-input-number t-size-s"
    modelvalue="3"
  >
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__decrease"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-remove t-size-s"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128 480L896 480 896 544 128 544Z"
          fill-opacity=".9"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-input-number-text-align"
        placeholder=""
      />
    </div>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__increase"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-add t-size-s"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M544 160L544 480 864 480 864 544 544 544 544 864 480 864 480 544 160 544 160 480 480 480 480 160 544 160Z"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
  </div>
  <div
    class="t-input-number t-size-m"
    modelvalue="3"
    style="margin: 0px 50px;"
  >
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__decrease"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-remove t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128 480L896 480 896 544 128 544Z"
          fill-opacity=".9"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-input-number-text-align"
        placeholder=""
      />
    </div>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__increase"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-add t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M544 160L544 480 864 480 864 544 544 544 544 864 480 864 480 544 160 544 160 480 480 480 480 160 544 160Z"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
  </div>
  <div
    class="t-input-number t-size-l"
    modelvalue="3"
  >
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__decrease"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-remove t-size-l"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128 480L896 480 896 544 128 544Z"
          fill-opacity=".9"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-input-number-text-align"
        placeholder=""
      />
    </div>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__increase"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-add t-size-l"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M544 160L544 480 864 480 864 544 544 544 544 864 480 864 480 544 160 544 160 480 480 480 480 160 544 160Z"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
  </div>
</div>
`;

exports[`InputNumber step demo works fine 1`] = `
<div>
  <div
    class="t-input-number t-size-m"
    modelvalue="3.2"
  >
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__decrease"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-remove t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128 480L896 480 896 544 128 544Z"
          fill-opacity=".9"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-input-number-text-align"
        placeholder=""
      />
    </div>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__increase"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-add t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M544 160L544 480 864 480 864 544 544 544 544 864 480 864 480 544 160 544 160 480 480 480 480 160 544 160Z"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
  </div>
</div>
`;

exports[`InputNumber undefined demo works fine 1`] = `
<div>
  <div
    class="t-input-number t-size-m"
  >
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__decrease"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-remove t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128 480L896 480 896 544 128 544Z"
          fill-opacity=".9"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-input-number-text-align"
        placeholder="输入"
      />
    </div>
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-button--icon-only t-input-number__increase"
      type="button"
    >
      
      
      <svg
        class="t-icon t-icon-add t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M544 160L544 480 864 480 864 544 544 544 544 864 480 864 480 544 160 544 160 480 480 480 480 160 544 160Z"
          fill-rule="evenodd"
        />
      </svg>
      
      
      
    </button>
  </div>
</div>
`;
